cover_agent/UnitTestGenerator.py

```diff
@@ -292,14 +293,32 @@ def validate_test(self, generated_test: dict, generated_tests_dict: dict):
             # Step 1: Append the generated test to the relevant line in the test file
             with open(self.test_file_path, "r") as test_file:
                 original_content = test_file.read()  # Store original content
+            # Need to understand the language used to perform the test.
+            test_language = self.get_code_language(self.test_file_path)
+
             original_content_lines = original_content.split("\n")
             test_code_lines = test_code_indented.split("\n")
-            # insert the test code at the relevant line
-            processed_test_lines = original_content_lines[:relevant_line_number_to_insert_after] + test_code_lines + original_content_lines[relevant_line_number_to_insert_after:]
-            processed_test = "\n".join(processed_test_lines)
-            # insert the additional imports at the top of the file
-            if additional_imports and additional_imports.rstrip() not in processed_test:
-                processed_test = additional_imports.rstrip() + "\n\n" + processed_test
+            processed_test = ''
+
+            if test_language == "groovy" or test_language == "java":
+                # Find the last closing brace and insert the processed_test before it
+                last_brace_index = original_content.rfind("}")
+                if last_brace_index != -1:
+                    processed_test = (
+                            original_content[:last_brace_index]
+                            + "\n"
+                            + test_code_indented
+                            + "\n"
+                            + original_content[last_brace_index:]
+                    )
+            else:
+                # insert the test code at the relevant line
+                processed_test_lines = original_content_lines[:relevant_line_number_to_insert_after] + test_code_lines + original_content_lines[relevant_line_number_to_insert_after:]
+                processed_test = "\n".join(processed_test_lines)
+                # insert the additional imports at the top of the file
+                if additional_imports and additional_imports.rstrip() not in processed_test:
+                    processed_test = additional_imports.rstrip() + "\n\n" + processed_test
+
```

Comment:
@mrT23 Is this something you had planned on implementing (or implemented)? Specifically the indented test code...

Reply:
after your refactor and it appears the new way you are inserting into the test file it is not no longer needed.